p2p/test/quic: race condition in TestQUICAndWebTransport

https://github.com/libp2p/go-libp2p/actions/runs/4121247736/jobs/7116750392

```
=== RUN   TestQUICAndWebTransport
  2023-02-08T06:11:24.803Z	ERROR	swarm2	swarm/swarm.go:232	error when shutting down connection: stream 0 canceled by remote with error code 1337
  ==================
  WARNING: DATA RACE
  Read at 0x00c000796160 by goroutine 90:
    github.com/quic-go/webtransport-go.(*Session).AcceptStream()
        /home/runner/go/pkg/mod/github.com/quic-go/webtransport-go@v0.5.0/session.go:242 +0x236
    github.com/libp2p/go-libp2p/p2p/transport/webtransport.(*conn).AcceptStream()
        /home/runner/work/go-libp2p/go-libp2p/p2p/transport/webtransport/conn.go:56 +0x68
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Conn).start.func1()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm_conn.go:114 +0x15d
  
  Previous write at 0x00c000796160 by goroutine 72:
    github.com/quic-go/webtransport-go.(*Session).CloseWithError()
        /home/runner/go/pkg/mod/github.com/quic-go/webtransport-go@v0.5.0/session.go:377 +0x144
    github.com/libp2p/go-libp2p/p2p/transport/webtransport.(*conn).Close()
        /home/runner/work/go-libp2p/go-libp2p/p2p/transport/webtransport/conn.go:72 +0x84
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Conn).doClose()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm_conn.go:75 +0x2fa
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Conn).doClose-fm()
        <autogenerated>:1 +0x39
    sync.(*Once).doSlow()
        /opt/hostedtoolcache/go/1.18.10/x64/src/sync/once.go:68 +0x101
    sync.(*Once).Do()
        /opt/hostedtoolcache/go/1.18.10/x64/src/sync/once.go:59 +0x46
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Conn).Close()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm_conn.go:58 +0x69
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Swarm).close.func2()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm.go:231 +0x8b
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Swarm).close.func5()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm.go:234 +0x47
  
  Goroutine 90 (running) created at:
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Conn).start()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm_conn.go:109 +0x8e
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Swarm).addConn()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm.go:339 +0x6fe
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Swarm).AddListenAddr.func2.2()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm_listen.go:141 +0xda
  
  Goroutine 72 (finished) created at:
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Swarm).close()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm.go:230 +0x7c4
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Swarm).close-fm()
        <autogenerated>:1 +0x39
    sync.(*Once).doSlow()
        /opt/hostedtoolcache/go/1.18.10/x64/src/sync/once.go:68 +0x101
    sync.(*Once).Do()
        /opt/hostedtoolcache/go/1.18.10/x64/src/sync/once.go:59 +0x46
    github.com/libp2p/go-libp2p/p2p/net/swarm.(*Swarm).Close()
        /home/runner/work/go-libp2p/go-libp2p/p2p/net/swarm/swarm.go:199 +0x53
    github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).Close.func1()
        /home/runner/work/go-libp2p/go-libp2p/p2p/host/basic/basic_host.go:996 +0x24d
    sync.(*Once).doSlow()
        /opt/hostedtoolcache/go/1.18.10/x64/src/sync/once.go:68 +0x101
    sync.(*Once).Do()
        /opt/hostedtoolcache/go/1.18.10/x64/src/sync/once.go:59 +0x46
    github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).Close()
        /home/runner/work/go-libp2p/go-libp2p/p2p/host/basic/basic_host.go:973 +0x50
    github.com/libp2p/go-libp2p/p2p/test/quic_test.TestQUICAndWebTransport.func2()
        /home/runner/work/go-libp2p/go-libp2p/p2p/test/quic/quic_test.go:127 +0x48
    runtime.deferreturn()
        /opt/hostedtoolcache/go/1.18.10/x64/src/runtime/panic.go:436 +0x32
    testing.tRunner()
        /opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1439 +0x213
    testing.(*T).Run.func1()
        /opt/hostedtoolcache/go/1.18.10/x64/src/testing/testing.go:1486 +0x47
  ==================
      testing.go:1312: race detected during execution of test
  --- FAIL: TestQUICAndWebTransport (0.37s)

--- a/go.mod
+++ b/go.mod
@@ -44,7 +44,7 @@ require (
 	github.com/pbnjay/memory v0.0.0-20210728143218-7b4eea64cf58
 	github.com/prometheus/client_golang v1.14.0
 	github.com/quic-go/quic-go v0.32.0
-	github.com/quic-go/webtransport-go v0.5.0
+	github.com/quic-go/webtransport-go v0.5.1
 	github.com/raulk/go-watchdog v1.3.0
 	github.com/stretchr/testify v1.8.1
 	go.uber.org/fx v1.18.2

--- a/test-plans/go.mod
+++ b/test-plans/go.mod
@@ -77,7 +77,7 @@ require (
 	github.com/quic-go/qtls-go1-19 v0.2.0 // indirect
 	github.com/quic-go/qtls-go1-20 v0.1.0 // indirect
 	github.com/quic-go/quic-go v0.32.0 // indirect
-	github.com/quic-go/webtransport-go v0.5.0 // indirect
+	github.com/quic-go/webtransport-go v0.5.1 // indirect
 	github.com/raulk/go-watchdog v1.3.0 // indirect
 	github.com/spaolacci/murmur3 v1.1.0 // indirect
 	go.uber.org/atomic v1.10.0 // indirect

